### ISSUE TITLE
How to save as TensorRT .engine or .plan file?

### ISSUE THREAD

**Issue by Hiperdyne19012** [NVIDIA-AI-IOT/torch2trt#254]
I want to use DeepStream with the pytorch model given in trt-pose. But I see no way to save the model as .engine or .plan. DeepStream does not support Pytorch model yet.

**Comment by anuar12** [comment]
It's saved as `state_dict` pytorch's object. You can access engine bytearray as follows:
```     def _create_engine(model_path: str): 
        state_dict = torch.load(model_path, map_location='cpu')
        engine_bytes = state_dict['engine']
        with trt.Runtime(trt_logger) as runtime:
            return runtime.deserialize_cuda_engine(engine_bytes)```

**Comment by Hiperdyne19012** [comment]
> It's saved as `state_dict` pytorch's object. You can access engine bytearray as follows:
> 
> ```
> def _create_engine(model_path: str): 
>         state_dict = torch.load(model_path, map_location='cpu')
>         engine_bytes = state_dict['engine']
>         with trt.Runtime(trt_logger) as runtime:
>             return runtime.deserialize_cuda_engine(engine_bytes)```
> ```

Thanks, I will try it out and let you know.

**Comment by ma-siddiqui** [comment]
Hi,

In my case, engine is created and loaded successfully but there is no input names. Only output names are shown. I cannot find input nodes. Any comment.

**Comment by ishangupta3** [comment]
@ma-siddiqui were you able to solve this problem?

**Comment by ma-siddiqui** [comment]
Yes. Input node names were resolved  but faced bad accuracy. After conversion it is dropped to 0.5%. Now struggling for it.

**Comment by ishangupta3** [comment]
@ma-siddiqui thanks for the reply, are you running on jetson Nano?

**Comment by ma-siddiqui** [comment]
At the moment I have DGX server but in future we have plan to take such a route based on business need. I used deep stream but this model conversion was planned to deploy on DGX server using trt inference server,  not for deep stream.

**Comment by ishangupta3** [comment]
Understood, thanks @ma-siddiqui

**Comment by HollrayChan** [comment]
save as an engine, I hope it can help :)

```
data = torch.ones((args.max_batch_size, 3, args.height, args.width)).cuda()
model_trt = torch2trt.torch2trt(model, [data],
                                fp16_mode=fp16,
                                int8_mode=int8,
                                max_workspace_size=args.max_workspace_size,
                                max_batch_size=args.max_batch_size)

engine_path = 'wow.engine'
with open(engine_path, "wb") as f:
    f.write(model_trt.engine.serialize())
```

**Comment by leaf918** [comment]
> Understood, thanks @ma-siddiqui

Do you have the torch2trt model work well in deepstream?
